contract/src/lib.rs

```diff
@@ -404,6 +409,69 @@ impl BtcLightClient {
         }
     }
 
+    fn check_target(&self, block_header: &Header, prev_block_header: &ExtendedHeader) {
+        if prev_block_header.block_height % BLOCKS_PER_ADJUSTMENT != 0 {
+            require!(
+                block_header.bits == prev_block_header.block_header.bits,
+                format!(
+                    "Error: Incorrect bits. Expected bits: {}; Actual bits: {}.",
+                    prev_block_header.block_header.bits, block_header.bits
+                )
+            );
+            return;
+        }
+
+        if prev_block_header.block_height + 1 >= BLOCKS_PER_ADJUSTMENT {
+            if let Some(init_header_hash) = self
```

Comment:
it will be better to rename it to `interval_tail_header_hash`
also maybe it is also good to rename the `mainchain_initial_blockhash` `main_chain_tail_block_hash`

Reply:
Also, we should panic here

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/64/commits/c1bfd8232cdc8f06ebf533fe6b43f69ef716c556

We should not panic here. It is normal case in the begin, when we don't have all the blocks, but already should update target.

Reply:
Maybe we should just avoid this scenario by forcing the developer initialising the contract from block multiple of 2016?

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/64/commits/c6215e245488db69dcd12516a810e1ae3696b394